projects/packages/connection/src/class-manager.php

```diff
@@ -1916,6 +1916,8 @@ public function get_authorization_url( $user = null, $redirect = null ) {
 				'site_created'          => $this->get_assumed_site_creation_date(),
 				'allow_site_connection' => ! $this->has_connected_owner(),
 				'calypso_env'           => ( new Host() )->get_calypso_env(),
+				// phpcs:disable WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+				'source'                => isset( $_GET['source'] ) ? wp_unslash( $_GET['source'] ) : '',
```

Comment:
For some other parameters, like the calypso env above, we rely on an allowlist of different options instead of allowing just about any string. Would it make sense here as well?

Reply:
Thanks for the review, @jeherve!

> For some other parameters, like the calypso env above, we rely on an allowlist of different options instead of allowing just about any string. Would it make sense here as well?

Agree. I have updated it now.